packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -102,10 +142,239 @@ class CupertinoContextMenu extends StatefulWidget {
+    )
+    this.previewBuilder = _defaultPreviewBuilder,
   }) : assert(actions != null && actions.isNotEmpty),
-       assert(child != null);
+       assert(child != null),
+       builder = ((BuildContext context, Animation<double> animation) => child);
+
+  /// Creates a  context menu with a custom [builder] controlling the widget.
+  ///
+  /// Use instead of the default constructor when it is needed to have a more
+  /// custom animation.
+  ///
+  /// [actions] is required and cannot be null or empty.
+  ///
+  /// [builder] is required.
+  CupertinoContextMenu.builder({
+    super.key,
+    required this.actions,
+    required this.builder,
+  }) : assert(actions != null && actions.isNotEmpty),
+       child = null,
+       previewBuilder = null;
+
+  /// Exposes the default border radius for matching iOS 16.0 behavior. This
+  /// value was eyeballed from the iOS simulator running iOS 16.0.
+  ///
+  /// {@tool snippet}
+  ///
+  /// Below is example code in order to match the default border radius for an
+  /// iOS 16.0 open preview.
+  ///
+  /// ```dart
+  /// CupertinoContextMenu.builder(
+  ///   actions: <Widget>[
+  ///     CupertinoContextMenuAction(
+  ///       child: const Text('Action one'),
+  ///       onPressed: () {},
+  ///     ),
+  ///   ],
+  ///   builder:(BuildContext context, Animation<double> animation) {
+  ///     final Animation<BorderRadius?> borderRadiusAnimation = BorderRadiusTween(
+  ///       begin: BorderRadius.circular(0.0),
+  ///       end: BorderRadius.circular(CupertinoContextMenu.kOpenBorderRadius),
+  ///     ).animate(
+  ///       CurvedAnimation(
+  ///         parent: animation,
+  ///         curve: Interval(
+  ///           CupertinoContextMenu.animationOpensAt,
+  ///           1.0,
+  ///         ),
+  ///       ),
+  ///     );
+  ///
+  ///     final Animation<Decoration> boxDecorationAnimation = DecorationTween(
+  ///       begin: const BoxDecoration(
+  ///        color: Color(0xFFFFFFFF),
+  ///        boxShadow: <BoxShadow>[],
+  ///       ),
+  ///       end: const BoxDecoration(
+  ///        color: Color(0xFFFFFFFF),
+  ///        boxShadow: CupertinoContextMenu.kEndBoxShadow,
+  ///       ),
+  ///      ).animate(
+  ///        CurvedAnimation(
+  ///         parent: animation,
+  ///         curve: Interval(
+  ///           0.0,
+  ///           CupertinoContextMenu.animationOpensAt,
+  ///       ))
+  ///     );
+  ///
+  ///     return Container(
+  ///       decoration:
+  ///         animation.value < CupertinoContextMenu.animationOpensAt ? boxDecorationAnimation.value : null,
+  ///       child: FittedBox(
+  ///         fit: BoxFit.cover,
+  ///         child: ClipRRect(
+  ///           borderRadius: borderRadiusAnimation.value ?? BorderRadius.circular(0.0),
+  ///           child: SizedBox(
+  ///             height: 150,
+  ///             width: 150,
+  ///             child: Image.network('https://picsum.photos/250?image=9'),
```

Comment:
Nit: Is there something else you could use besides this third party image url?  I just worry about the dependency.  I'm not sure if we have anything like this elsewhere in the docs or not.

Reply:
I'll switch to an image we have in the flutter assets for api docs folder we have on github, like what's used elsewhere in the docs.